builds.sh

```diff
@@ -130,11 +147,18 @@ for sourceId; do
 				end
 			')"
 
+			if [ -v "imageArchResolved["$lookup"]" ]; then
+				remoteArches="${imageArchResolved["$lookup"]}"
```

Comment:
```suggestion
				remoteArches=${imageArchResolved[$lookup]}
```

There is no need to quote any of this. Word splitting is not done on the right side of assignments.

Reply:
The quoting here doesn't cause any change in behavior and ensures we keep up on our quoting discipline more consistently, so I'd prefer to keep these types of assignments quoted.

Reply:
My concern here is how Bash deals with the nested double quotes. Does it _do the right thing™_? I know it does for sub-shells and it does not for single quotes, but I'm not sure about inside `${}` and it's difficult to tell since the square brackets effectively behave as quotes, i.e., the index can have spaces and this is perfectly safe: `${a[b c]}`. Interesting to note, I've seen constructs like this elsewhere in the code base: `v='w'$x'y'`, i.e., they leverage the fact that the right-hand side of the assignment operator do not undergo word splitting. So perhaps `"${imageArchResolved[$lookup]}"` is the cleanest option?

Reply:
If there's anywhere I haven't consistently written `v='w'$x'y'` as `v='w'"$x"'y'`, we should fix it.

Each of `foo[$bar]=...` vs `foo["$bar"]=...` and `"${foo[$bar]}"` vs `"${foo["$bar"]}"` do indeed work interchangeably, and do indeed work correctly (even when `$bar` contains multiple successive whitespaces):

```console
$ declare -A foo=()
$ bar='a   b' # note the three spaces
$ foo[$bar]=xyz
$ foo["$bar"]=abc
$ echo "${!foo[@]}"
a   b
$ echo "${foo[$bar]}"
abc
$ echo "${foo["$bar"]}"
abc
```

Reply:
Ah, `v='w'"$x"'y'` is probably what I was thinking of. Sorry, didn't mean to impugn your work.